src/main/java/net/starschema/clouddb/jdbc/BQForwardOnlyResultSet.java

```diff
@@ -249,12 +248,44 @@ private Timestamp toTimestamp(String value, Calendar cal) throws SQLException {
             return new Timestamp(cal.getTime().getTime());
         }
         catch (NumberFormatException e) {
-            throw new BQSQLException(e);
+            // before giving up, check to see if we've been given a 'time' value without a
+            // date, e.g. from current_time(), and if we have, try to parse it
+            try {
+                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+                java.util.Date parsedDate = dateFormat.parse(buildFullDateStringFromTime(value));
+                return new java.sql.Timestamp(parsedDate.getTime());
+            }
+            catch (Exception e2) {
+                throw new BQSQLException(e);
```

Comment:
I was looking for a way to catch exceptions anonymously in Java but I couldn't find one. 

If the outer/earlier `try` block in this method fails with a `NumberFormatException e`, and then we attempt to salvage by parsing the input string as a dateless timestamp and fail, I was thinking we'd want to surface the outer `try`'s exception (`e`) instead of the possibly-likely-to-fail inner one's exception (imaginatively named `e2`). 

Is there a way to
```java
try {}
catch {}
```
?? I didn't think there was